File: read-excel.js
const XLSX = require('xlsx');
const path = require('path');

const filePath = '/home/node/.openclaw/media/inbound/9708932a-b892-4147-954d-2a4a40f2abce.xlsx';

try {
    // Read the workbook
    const workbook = XLSX.readFile(filePath);
    
    console.log('=== Excel 文件信息 ===');
    console.log('工作表列表:', workbook.SheetNames);
    console.log('');
    
    // Read each sheet
    workbook.SheetNames.forEach(sheetName => {
        console.log(`=== 工作表：${sheetName} ===`);
        const worksheet = workbook.Sheets[sheetName];
        
        // Convert to JSON
        const jsonData = XLSX.utils.sheet_to_json(worksheet, {header: 1});
        
        // Print all rows
        jsonData.forEach((row, index) => {
            console.log(`行 ${index + 1}:`, row);
        });
        console.log('');
    });
    
    // Extract all text content to find names
    console.log('=== 提取所有文本内容 ===');
    const allText = [];
    workbook.SheetNames.forEach(sheetName => {
        const worksheet = workbook.Sheets[sheetName];
        const jsonData = XLSX.utils.sheet_to_json(worksheet, {header: 1});
        jsonData.forEach(row => {
            if (Array.isArray(row)) {
                row.forEach(cell => {
                    if (cell && typeof cell === 'string') {
                        allText.push(cell);
                    }
                });
            }
        });
    });
    
    console.log('所有文本内容:', allText);
    
} catch (error) {
    console.error('读取 Excel 文件时出错:', error);
}

File: read-excel.py
import pandas as pd
import sys

file_path = '/home/node/.openclaw/media/inbound/9708932a-b892-4147-954d-2a4a40f2abce.xlsx'

try:
    # Read all sheets
    xl = pd.ExcelFile(file_path)
    sheet_names = xl.sheet_names
    
    print('=== Excel 文件信息 ===')
    print(f'工作表列表：{sheet_names}')
    print('')
    
    all_names = []
    all_text = []
    
    for sheet_name in sheet_names:
        print(f'=== 工作表：{sheet_name} ===')
        df = pd.read_excel(file_path, sheet_name=sheet_name)
        
        # Print all data
        print(df.to_string())
        print('')
        
        # Extract all text values
        for col in df.columns:
            for val in df[col].dropna():
                val_str = str(val)
                all_text.append(val_str)
    
    print('=== 所有文本内容 ===')
    print(all_text)
    print('')
    
    # Look for potential names (Chinese characters patterns)
    import re
    chinese_pattern = re.compile(r'[\u4e00-\u9fff]+')
    
    potential_names = []
    for text in all_text:
        # Find Chinese character sequences
        matches = chinese_pattern.findall(str(text))
        for match in matches:
            if len(match) >= 2:  # At least 2 Chinese characters
                potential_names.append(match)
    
    print('=== 潜在人名（中文）===')
    print(list(set(potential_names)))
    
except Exception as e:
    print(f'读取 Excel 文件时出错：{e}')
    import traceback
    traceback.print_exc()
